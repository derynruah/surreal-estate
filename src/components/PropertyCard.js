import React from "react";
import "../styles/property-card.css";

const PropertyCard = ({
  title,
  type,
  bathrooms,
  bedrooms,
  price,
  city,
  email,
}) => (
  <div className="property-card">
    <div className="detail">
      <h3>{title}</h3>
    </div>
    <div className="detail">
      <em>
        {type} - {city}
      </em>
    </div>
    <div className="detail">
      <i className="fas fa-bath" /> {bathrooms}
    </div>
    <div className="detail">
      <i className="fas fa-bed" /> {bedrooms}
    </div>
    <div className="detail">
      <i className="fas fa-pound-sign" /> {price}
    </div>
    <a href={`mailto:${email}`} className="email">
      <i className="fas fa-envelope" /> Email
    </a>
  </div>
);

export default PropertyCard;
