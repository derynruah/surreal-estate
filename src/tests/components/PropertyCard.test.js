import React from "react";
import { render, screen } from "@testing-library/react";
import PropertyCard from "../../components/PropertyCard";

describe("PropertyCard", () => {
  const validProps = {
    title: "Stub description",
    type: "Stub type",
    bathrooms: 2,
    bedrooms: 5,
    price: 10000,
    city: "Stub city",
    email: "stub@email.com",
  };

  test("renders correctly", () => {
    const { asFragment } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("renders correct values for props", () => {
    render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />
    );

    expect(screen.getByText(validProps.title)).toBeInstanceOf(
      HTMLHeadingElement
    );
    expect(screen.getByText(new RegExp(validProps.type))).toHaveTextContent(
      `${validProps.type} - ${validProps.city}`
    );
    expect(screen.getByText(validProps.bathrooms)).toHaveTextContent(/2/i);
    expect(screen.getByText(validProps.bedrooms)).toHaveTextContent(/5/i);
    expect(screen.getByText(validProps.price)).toHaveTextContent("10000");
    expect(screen.getByText(new RegExp(validProps.city))).toHaveTextContent(
      `${validProps.type} - ${validProps.city}`
    );
    expect(screen.getByText(/email/i)).toHaveAttribute(
      "href",
      `mailto:${validProps.email}`
    );
  });
});
