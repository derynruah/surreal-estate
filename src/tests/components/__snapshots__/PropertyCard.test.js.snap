// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <div
      class="detail"
    >
      <h3>
        Stub description
      </h3>
    </div>
    <div
      class="detail"
    >
      <em>
        Stub type - Stub city
      </em>
    </div>
    <div
      class="detail"
    >
      <i
        class="fas fa-bath"
      />
       2
    </div>
    <div
      class="detail"
    >
      <i
        class="fas fa-bed"
      />
       5
    </div>
    <div
      class="detail"
    >
      <i
        class="fas fa-pound-sign"
      />
       10000
    </div>
    <a
      class="email"
      href="mailto:stub@email.com"
    >
      <i
        class="fas fa-envelope"
      />
       Email
    </a>
  </div>
</DocumentFragment>
`;
